pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -502,6 +502,14 @@ func (c *Cluster) getOSDInfo(d *appsv1.Deployment) (OSDInfo, error) {
 		}
 	}
 
+	// if the ROOK_TOPOLOGY_AFFINITY env var was not found in the loop above, detect it from the node
+	if isPVC && osd.TopologyAffinity == "" {
+		osd.TopologyAffinity, err = getTopologyFromNode(c.context.Clientset, d, osd)
+		if err != nil {
+			logger.Errorf("failed to get topology affinity for osd %d", osd.ID)
```

Comment:
```suggestion
			logger.Errorf("failed to get topology affinity for osd %d. %v", osd.ID, err)
```

It is missing the error.